Update hip-584 (#625)
hashgraph/hedera-improvement-proposal: HIP/hip-584.md

#### Read-only queries and gas estimate REST API

A new  `/api/v1/contracts/call` POST REST API will be added to execute read-only queries and gas estimate.

> Query parameter `estimate`(default = false) would determine what the result represents.

Reviewer: I do still think there will be a default of false even if it's moved into the body and that should be mentioned here.

> Body parameter `estimate` would determine what the result represents.

Reviewer: Body parameter terminology is a bit confusing.

```suggestion
The `estimate` field in the request body would determine what the result represents.
```

  "gas": "0x76c0", 
  "gas_price": "0x76c0", 
  "to": "0x00000000000000000000000000000000000004e4", 
  "value": "0x0",

>   "estimate": "true"

Reviewer: Use boolean type. Also sort with other fields.
```suggestion
  "estimate": true
```

}
```
The following JSON represents a typical response result with the information read in the query:  

> With body parameter `estimate=true`

Reviewer: ```suggestion
With the `estimate` field set to `true` in the request body:
```

```json
{
    "gas": "0x3234333230",
    "result": "0x"
}
```

> With body parameter `estimate=false`

Reviewer: ```suggestion
With the `estimate` field omitted or set to `false` in the request body:
```
